### PR TITLE
Implement GitHub PR Issue Checker Workflow

### LINKED ISSUE #900: Implement PR Issue Checker Workflow
**Description:** We need to create a GitHub Actions workflow that automates the validation of pull requests (PRs). This workflow will ensure that all PRs meet the following criteria before being merged:

**PR Description Check:** Every PR must have a description. If the description is missing, the PR should fail the check.
    
**Issue Reference Check:** The PR description must include an issue reference in the format Fixes #<issue-number>. If the PR is not addressing an existing issue, it should mention Fixes #NEW as a placeholder.

**Acceptance Criteria:**

- A GitHub Action workflow is triggered on PR events (opened, edited).

**The workflow checks the PR body for:**
- A non-empty description.
- The presence of Fixes #<issue-number> or Fixes #NEW.

**If the criteria are not met:**

- The workflow should fail and return clear error messages.

**If the checks pass:**
- A confirmation message is outputted.

**Expected Output:**

- Error message if the PR description is missing.
- Error message if the issue reference is missing or incorrectly formatted.
- Success message when all checks are satisfied.

**Additional Notes:**

- This issue may serve as a foundation for more advanced PR validation (e.g., commit message format, label enforcement).
- We can later extend this to check for specific branch naming conventions or enforce the addition of specific labels.

_**Note**_: i'm expecting a gssoc-ext with level3 (or) level2 and a hacktoberfest-accepted label!

@ajay-dhangar

### PR DESCRIPTION
**Pull request by smog-root**
**Fixes:** #900 

**Usage:**

1. When a pull request is created or edited, the workflow automatically triggers.

2. The workflow checks for:
- A non-empty PR description.
- A valid issue reference format (`Fixes #<issue-number>` or `Fixes #NEW`).
3. If either check fails, the PR will be marked as failing, and the author will receive clear error messages in the PR comments.
4. If all checks pass, a confirmation message will indicate that the PR is valid.

**_Note_**: Kindly add all the labels added to the issue, to this PR!

### CODE REVIEW

File: .github/workflows/pr-checker.yml [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+        run: |
+          PR_DESCRIPTION="${{ github.event.pull_request.body }}"
+          if [[ -z "$PR_DESCRIPTION" ]]; then
+            echo "PR description is missing."
+            exit 1
+          fi
+
+          if [[ ! "$PR_DESCRIPTION" =~ Fixes\ #[0-9]+ ]]; then
+            echo "The PR description should include 'Fixes #<issue-number>' if not addressing any issue."
+            echo "##[error]Fixes #NEW must be included in the description."
```
Comment (ajay-dhangar) [inline .github/workflows/pr-checker.yml]: replace `#NEW` with `#ISSUE_NUMBER`

**Review by ajay-dhangar — CHANGES_REQUESTED** [review]
Thank you for your pull request. Please make only one line change, and then I will merge it.

File: .github/workflows/pr-checker.yml (hunk 1 above)
Comment (smog-root) [reply .github/workflows/pr-checker.yml]: Done 👍!

**Comment by smog-root** [comment]
i have updated the changes! merge it! @ajay-dhangar

**Comment by smog-root** [comment]
hey @ajay-dhangar in the issue #900 .You added level2 label. 
But now in the PR you added level1! please don't do that

**Review by ajay-dhangar — APPROVED** [review]
Done

**Comment by smog-root** [comment]
Add hacktoberfest-accepted label! @ajay-dhangar

**Pull request merged**